### PR TITLE
feat: Upgrade swc_* related crates

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
LGTM

**Review by dsherret — APPROVED** [review]

File: src/transpiling/jsx_precompile.rs [hunk 1]
```diff
@@ -2135,7 +2135,7 @@ const a = _jsxTemplate($$_tpl_1);"#,
 </div>;"#,
       r#"import { jsxTemplate as _jsxTemplate } from "react/jsx-runtime";
 const $$_tpl_1 = [
-  "<div>foo bar</div>"
+  "<div>foo bar		</div>"
```
Comment (bartlomieju) [inline src/transpiling/jsx_precompile.rs:2138]: @marvinhagemeister is this expected?

File: src/transpiling/jsx_precompile.rs [hunk 2]
```diff
@@ -2149,7 +2149,7 @@ const result = _jsxTemplate($$_tpl_1);"#,
 </div>;"#,
       r#"import { jsxTemplate as _jsxTemplate } from "react/jsx-runtime";
 const $$_tpl_1 = [
-  "<div>foo bar</div>"
+  "<div>foo bar    </div>"
```
Comment (bartlomieju) [inline src/transpiling/jsx_precompile.rs:2152]: Ditto

File: src/transpiling/jsx_precompile.rs (hunk 1 above)
Comment (dsherret) [reply src/transpiling/jsx_precompile.rs:2138]: Maybe has something to do with https://github.com/dprint/dprint-swc-ext/pull/61/files#diff-2ef47ffbd458b1e790485c74567c004f2ce9c7081d92cea8c5d9acfdebd5ef48R17643-R17644

Comment (bartlomieju) [reply src/transpiling/jsx_precompile.rs:2138]: I think there was a change in SWC in how `JSXText` is handled now and the condition for `lines.peek().is_some()` in `jsx_test_to_str` is no longer valid. I'll wait for Marvin's input how we should handle that.

Comment (marvinhagemeister) [reply src/transpiling/jsx_precompile.rs]: I've pushed a fix for this by improving the logic in the precompile transform that decides when trailing whitespace of a `JSXText` element can be safely trimmed or not. The resulting behavior should be the exact same as before the JSX update and makes the transform more robust.

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**